### PR TITLE
Feat real time data

### PR DESCRIPTION
**Pull request by kyrel**

### CODE REVIEW

File: src/plugins/load-spaces-data.server.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useSpacesStore } from "~/stores/spaces";
+
+export default defineNuxtPlugin((app) => {
+  const spacesStore = useSpacesStore(app.$pinia);
+  import("~/data/buildings.json").then((buildings) => {
+    spacesStore.setBuildings(buildings.default as any); //TODO: arrays can't become tuples, silly things
+  });
+  import("~/data/spaces.json").then((spaces) => {
+    spacesStore.setSpaces(spaces.default as any); //TODO: arrays can't become tuples, silly things
```
Comment (Siilwyn) [inline src/plugins/load-spaces-data.server.ts:9]: Todo

File: src/stores/spaces.ts [hunk 2]
```diff
@@ -56,7 +62,18 @@ export const useStore = defineStore("index", () => {
...
+  function bulkSetBuildingOccupancy(data: Record<number, number>) {
+    buildingsI18n.value = buildingsI18n.value.map((building) => {
+      return { ...building, activeDevices: data[building.number] };
+    });
+  }
+
+  function setBuildingOccupancy(buildingNumber: number, activeDevices: number) {
+    const building = buildingsI18n.value.find(
+      (building) => building.number === buildingNumber
+    ); //TODO: extract function back
```
Comment (Siilwyn) [inline src/stores/spaces.ts:74]: Todo

**Review by Siilwyn — APPROVED** [review]
LGTM but don't think we should add more todo comments, let's convert them to tickets?

File: src/types/supabase.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
+export type Json =
```
Comment (jbmoelker) [inline src/types/supabase.ts:1]: These types are great to have! Hope you didn't type them manually ;) I imagine you used the [Supabase CLI for generating TS types](https://supabase.com/docs/guides/database/api/generating-types). So I would add a prep script to our package to automate this and keep this in sync in the future.

**Review by jbmoelker — APPROVED** [review]
I like the improved code organisation including moving things to plugins and other helpers!

Would love the occupancy indicator to be displayed in [the building list](https://deploy-preview-143--spacefinder.netlify.app/en/buildings/) as that will give the most impressive demo effect I think. Something like adding **x devices** to cards so you get:

Aerospace Engineering (AE)
7 locations | 323 seats | **39 devices**

Applied Physics (AS)
34 locations | 866 seats | **948 devices**

etc

**Pull request merged**